src/Microsoft.ML.Tokenizers/Tokenizer.cs

```diff
@@ -324,6 +366,45 @@ private int LastIndexOf(string text, int maxTokenCount, out string processedText
         public static Tokenizer CreateTiktokenForModel(string modelName, IReadOnlyDictionary<string, int>? extraSpecialTokens = null, Normalizer? normalizer = null)
                         => Tiktoken.CreateTokenizerForModel(modelName, extraSpecialTokens, normalizer);
 
+        /// <summary>
+        /// Create tokenizer based on encoding name
+        /// </summary>
+        /// <param name="encodingName">Encoding name</param>
+        /// <param name="extraSpecialTokens">Extra special tokens other than the built-in ones for the encoding</param>
+        /// <param name="normalizer">To normalize the text before tokenization</param>
+        /// <returns>The tokenizer</returns>
+        public static Tokenizer CreateTiktokenForEncoding(string encodingName, IReadOnlyDictionary<string, int>? extraSpecialTokens = null, Normalizer? normalizer = null)
+        {
+            if (string.IsNullOrEmpty(encodingName))
+            {
+                throw new ArgumentNullException(nameof(encodingName));
+            }
+
+            string modelName;
+            if (encodingName.Equals(Tiktoken.Cl100kBaseEncodingName, StringComparison.OrdinalIgnoreCase))
+            {
+                modelName = "gpt-4";
+            }
```

Comment:
Why do we need this intermediate identifier?  Instead can you map these to the `ModelEncoding` type and then use an internal version of `CreateTokenizerForModel` that takes a `ModelEncoding`?

Here's a sketch: https://github.com/ericstj/machinelearning/commit/e8ea567ce40cef539dc74b1b8240c458bc52b0d0

Reply:
I don't have a strong feeling either way. I was trying to use minimum knowledge from Titoken internals (like ModelEncoding) in the Tokenizer class but that is not big deal. I'll do it your way.

Reply:
Thank you